pkg/services/authn/grpcutils/grpc_authenticator.go

```diff
@@ -19,101 +19,79 @@ import (
 	"github.com/grafana/grafana/pkg/setting"
 )
 
-var once sync.Once
-
 func NewInProcGrpcAuthenticator() interceptors.Authenticator {
+	return newAuthenticator(
+		authn.NewDefaultAuthenticator(
+			authn.NewUnsafeAccessTokenVerifier(authn.VerifierConfig{}),
+			authn.NewUnsafeIDTokenVerifier(authn.VerifierConfig{}),
+		),
+		tracing.NewNoopTracerService(),
+	)
+}
+
+func NewAuthenticator(cfg *GrpcServerConfig, tracer tracing.Tracer) interceptors.Authenticator {
+	client := http.DefaultClient
+	if cfg.AllowInsecure {
+		client = &http.Client{Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}}
+	}
+
+	kr := authn.NewKeyRetriever(authn.KeyRetrieverConfig{
+		SigningKeysURL: cfg.SigningKeysURL,
+	}, authn.WithHTTPClientKeyRetrieverOpt(client))
+
 	auth := authn.NewDefaultAuthenticator(
-		authn.NewUnsafeAccessTokenVerifier(authn.VerifierConfig{}),
-		authn.NewUnsafeIDTokenVerifier(authn.VerifierConfig{}),
+		authn.NewAccessTokenVerifier(authn.VerifierConfig{AllowedAudiences: cfg.AllowedAudiences}, kr),
+		authn.NewIDTokenVerifier(authn.VerifierConfig{}, kr),
 	)
 
+	return newAuthenticator(auth, tracer)
+}
+
+func NewAuthenticatorWithFallback(cfg *setting.Cfg, reg prometheus.Registerer, tracer tracing.Tracer, fallback interceptors.Authenticator) interceptors.Authenticator {
```

Comment:
There is no way to disable the fallback by config anymore. I think you should revert this change.

Reply:
Sorry missed that part